cosmwasm/packages/sgx-vm/src/wasmi/exports.rs

```diff
@@ -76,6 +71,58 @@ pub extern "C" fn ocall_read_db(
         .unwrap_or(OcallReturn::Panic)
 }
 
+/// Read a key from the contracts key-value store.
+#[no_mangle]
+pub extern "C" fn ocall_query_chain(
+    context: Ctx,
+    vm_error: *mut UntrustedVmError,
+    gas_used: *mut u64,
+    value: *mut EnclaveBuffer,
+    query: *const u8,
+    query_len: usize,
+) -> OcallReturn {
+    let query = unsafe { std::slice::from_raw_parts(query, query_len) };
+
+    let implementation = unsafe { get_implementations_from_context(&context).query_chain };
+
+    std::panic::catch_unwind(|| implementation(context, query))
+        // Get either an error(`OcallReturn`), or a response(`EnclaveBuffer`)
+        // which will be converted to a success status.
+        .map(|answer| -> Result<EnclaveBuffer, OcallReturn> {
+            match answer {
+                Ok((system_result, gas_cost)) => {
+                    unsafe { *gas_used = gas_cost };
+
+                    /*
+                        wasm code expects to get this as Result<Result<Binary, StdError>, SystemError> which is called SystemResult
+                        see CosmWasm's implementation https://github.com/enigmampc/SecretNetwork/blob/508e99c990dd656eb61f456584dab054487ba178/cosmwasm/packages/sgx-vm/src/imports.rs#L124
+                    */
+
+                    crate::serde::to_vec(&system_result)
```

Comment:
Can you please make this a regular `//` comment so it takes 2 lines instead of 5?